src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -48,34 +48,32 @@ internal TrainTestData(IDataView trainSet, IDataView testSet)
 
         /// <summary>
         /// Split the dataset into the train set and test set according to the given fraction.
-        /// Respects the <paramref name="stratificationColumn"/> if provided.
+        /// Respects the <paramref name="idColumn"/> if provided.
         /// </summary>
         /// <param name="data">The dataset to split.</param>
         /// <param name="testFraction">The fraction of data to go into the test set.</param>
-        /// <param name="stratificationColumn">Optional name of the column to use as a stratification column. If two examples share the same value of the <paramref name="stratificationColumn"/>
-        /// (if provided), they are guaranteed to appear in the same subset (train or test). Use this to make sure there is no label leakage from train to the test set.
-        /// If this optional parameter is not provided, a stratification columns will be generated, and its values will be random numbers .</param>
-        /// <param name="seed">Optional parameter used in combination with the <paramref name="stratificationColumn"/>.
-        /// If the <paramref name="stratificationColumn"/> is not provided, the random numbers generated to create it, will use this seed as value.
-        /// And if it is not provided, the default value will be used.</param>
-        public TrainTestData TrainTestSplit(IDataView data, double testFraction = 0.1, string stratificationColumn = null, uint? seed = null)
+        /// <param name="idColumn">Name of a column to use as an ID for grouping rows. If two examples share the same value of the <paramref name="idColumn"/>,
+        /// they are guaranteed to appear in the same subset (train or test). This can be used to ensure no label leakage from train to the test set.
+        /// If <see langword="null"/> no row grouping will be performed.</param>
+        /// <param name="seed">Seed for the random number generator used to select rows for the train-test split.</param>
```

Comment:
`seed` isn't actually doing this here. Looks like it is used only in `EnsureIdColumn`. Otherwise, we just return the first `testFraction` rows as the test set, and the complement as the train set. 

We should be doing train test splits with a seed where the split is random based on the seed provided, but given current behavior, the presence of this argument (or at least the description) would cause confusion to the user.